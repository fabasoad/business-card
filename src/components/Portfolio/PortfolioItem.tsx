import * as React from 'react'

interface PortfolioItemProps {
  img: any
  name: string
  subtitle: string
  title: string
  url: string
}

export default function PortfolioItem(props: PortfolioItemProps) {
  return (
    <div className="col-sm-6 col-md-3 col-lg-3 web">
      <div className="portfolio-item">
        <div className="hover-bg text-center">
          <a href={props.url} title={props.name} rel="noopener noreferrer" target="_blank">
            <div className="hover-text">
              <h4>{props.title}</h4>
              <small>{props.subtitle}</small>
              <i className="fa fa-search"></i>
            </div>
            <img className="img-thumbnail" src={props.img} alt={props.name} />
          </a>
        </div>
      </div>
    </div>
  )
}
