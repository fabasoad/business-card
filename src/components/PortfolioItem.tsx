import * as React from 'react';

export default function PortfolioItem(props) {
  return (
    <div className="col-sm-6 col-md-3 col-lg-3 web">
      <div className="portfolio-item">
        <div className="hover-bg text-center">
          <a href={props.url} title={props.name} target="_blank">
            <div className="hover-text">
              <h4>{props.title}</h4>
              <small>{props.subtitle}</small>
              <div className="clearfix"></div>
              <i className="fa fa-search"></i>
            </div>
            <img className="img-thumbnail" src={props.img} alt={props.name} />
          </a>
        </div>
      </div>
    </div>
  );
}