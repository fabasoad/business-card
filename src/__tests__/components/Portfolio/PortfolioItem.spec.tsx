/// <reference types="jest" />
import * as React from 'react'
import { shallow, ShallowWrapper } from 'enzyme'
import PortfolioItem from '../../../components/Portfolio/PortfolioItem'

test('should render PortfolioItem correctly', () => {
  const wrapper: ShallowWrapper = shallow(<PortfolioItem
    img="test-img"
    name="test-name"
    subtitle="test-subtitle"
    title="test-title"
    url="test-url"
  />)
  expect(wrapper).toMatchSnapshot()
})
