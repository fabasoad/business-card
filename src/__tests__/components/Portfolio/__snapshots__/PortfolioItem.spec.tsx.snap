// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render PortfolioItem correctly 1`] = `
<div
  className="col-sm-6 col-md-3 col-lg-3 web"
>
  <div
    className="portfolio-item"
  >
    <div
      className="hover-bg text-center"
    >
      <a
        href="test-url"
        rel="noopener noreferrer"
        target="_blank"
        title="test-name"
      >
        <div
          className="hover-text"
        >
          <h4>
            test-title
          </h4>
          <small>
            test-subtitle
          </small>
          <i
            className="fa fa-search"
          />
        </div>
        <img
          alt="test-name"
          className="img-thumbnail"
          src="test-img"
        />
      </a>
    </div>
  </div>
</div>
`;
